### PR TITLE
Add version field to default serialized format

### PR DESCRIPTION
**Pull request by JLockerman**
This commit adds a version byte, currently always set to 1, to the default serialization format. This will allow us to change the serialization format in a forwards compatible manner, by dispatching on this field.

### CODE REVIEW

File: extension/src/type_builder.rs [hunk 1]
```diff
@@ -150,13 +150,15 @@ macro_rules! do_serialize {
...
-            .unwrap_or_else(|e| pgx::error!("serialization error {}", e));
+                .unwrap_or_else(|e| pgx::error!("serialization error {}", e)) + 1;
             let mut bytes = Vec::with_capacity(size as usize + 4);
             let mut varsize = [0; 4];
             unsafe {
                 pgx::set_varsize(&mut varsize as *mut _ as *mut _, size as _);
             }
             bytes.extend_from_slice(&varsize);
+            // serialization version; 1 for bincode is currently the only option
+            bytes.push(1);
```
Comment (davidkohn88) [inline extension/src/type_builder.rs:161]: Why not make this a parameter of the function and have everything using it call with it?

**Review by davidkohn88 — COMMENTED** [review]
I'm not sure why we're setting this to 1 always now rather than making it settable by the calling function. Is the plan that we increment this version for everything all at once? Or that each type of thing will have its own versioning?

**Comment by JLockerman** [comment]
I expect the most likely scenario is that we never change the version number, and that the default should make this unchanging use case as easy as possible. I think the second most likely scenario is that we change our default serialization format for everything. In the event we do want change a specific one, we'll have to add an optional argument.

File: extension/src/type_builder.rs (hunk 1 above)
Comment (WireBaron) [inline extension/src/type_builder.rs:161]: It would be better if we used an enum here instead of a magic constant.

**Comment by JLockerman** [comment]
bors retry

**Pull request merged**